Add to the accessibility support section of the docs (#1730)
pydata/pydata-sphinx-theme: docs/user_guide/accessibility.md

- We have improved how we assign text colors to interactive elements such as buttons and dropdowns to ensure that they meet
  WCAG 2 AA or AAA contrast requirements.

> ### Keyboard Support

Reviewer: ```suggestion
### Keyboard Navigation Support
```

For all buttons, dropdowns, tabbed panels, hamburger menus, modals, overlays,
links and other interactive elements, we have worked to ensure they:

- Have a visible focus indicator (WCAG 2.4.7)

> - Work with a keyboard (WCAG 2.1.1)

Reviewer: ```suggestion
- Can be accessed via keyboard navigation (WCAG 2.1.1)
```
